WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -164,7 +164,9 @@ static IEnumerable<(int Size, FeeRate From, FeeRate To)> SplitFeeGroupInBlocks(F
 		// then we don't want our estimations to be affected by those rare cases.
 		var relevantFeeGroups = mempoolInfo.Histogram
 			.OrderByDescending(x => x.Group)
-			.SkipWhile(x => x.Count < mempoolInfo.Size / 1_000);
+			.SkipWhile((_, index) => mempoolInfo.Histogram.Take(index + 1).Sum(g => g.Count) < mempoolInfo.Size / 1_000)
```

Comment:
This is a `SELECT 1+N` algorithm what it is not problematic in terms of performance but it is hard to understand. Could you simplify the logic?